src/pymatgen/io/vasp/inputs.py

```diff
@@ -345,15 +346,25 @@ def from_str(
         else:
             lattice *= scale
 
-        vasp5_symbols: bool = False
+        # "atomic_symbols" is the "fully expanded" list of symbols,
+        # while "symbols" is the list as shown in POSCAR.
+        # For example with a POSCAR:
+        # ... (comment/scale/lattice)
+        # H O
+        # 2 1
+        # ...
+        # atomic_symbols is ["H", "H", "O"]
+        # symbols is ["H", "O"]
         atomic_symbols: list[str] = []
+        vasp5or6_symbols: bool = False
+        symbols: list[str] | None = None
 
         try:
             n_atoms: list[int] = [int(i) for i in lines[5].split()]
             ipos: int = 6
 
         except ValueError:
-            vasp5_symbols = True
```

Comment:
I guess this tag `vasp5_symbols` (added 2012) is used to differentiate VASP4 and VASP5/6? Instead of VASP 5 vs VASP 6?

![image](https://github.com/user-attachments/assets/a7f03497-a1dc-4a37-8e24-37dcf1db7ba1)

I have absolutely zero knowledge about VASP 4.x (seemingly before 2008), and didn't even manage to find a VASP 4 POSCAR.

Anyone with experience on VASP 4.x please free feel to comment :)